### PR TITLE
chore: get browser logs from failed cypress tests

### PR DESCRIPTION
**Pull request by akash-codemonk**
Fixes https://github.com/appsmithorg/appsmith/issues/26006

Get browser console logs for failed tests. This will help with easier debugging of flaky tests. 

Sample artifact for a test https://github.com/appsmithorg/appsmith/suites/16856147292/artifacts/962768313
Latest test run of the change: https://github.com/appsmithorg/appsmith/actions/runs/6405413689

Trimming down the files to a max of 10 files, since the artifact size can go upto a GB when there are a 1000 or more tests failing.

### CODE REVIEW

File: .github/workflows/ci-test-limited.yml [hunk 1]
```diff
@@ -521,6 +521,16 @@ jobs:
           config-file: cypress_ci_custom.config.ts
           working-directory: app/client
           env: "NODE_ENV=development"
+      
+      - name: Trim number of cypress log files
+        run: |
+          find ${{ github.workspace }}/app/client/cypress/cypress-logs -name '*.json' -type f | tail -n +11 | xargs -I {} rm -- {}
```
Comment (Aishwarya-U-R) [inline .github/workflows/ci-test-limited.yml:527]: What this does?

File: app/client/cypress/limited-tests.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
+#cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
 
 
 # For running all specs - uncomment below:
-#cypress/e2e/**/**/*
+cypress/e2e/**/**/*
```
Comment (Aishwarya-U-R) [inline app/client/cypress/limited-tests.txt:6]: FYI - This change should be reverted before PR can be merged.

File: .github/workflows/ci-test-limited.yml (hunk 1 above)
Comment (akash-codemonk) [reply .github/workflows/ci-test-limited.yml:527]: For each failed test it creates a file with logs. Here we are storing a maximum of 10 files, rest are deleted to keep the artifact size small. Also usually we know what is the cause of test failure is when a lot of tests are failing.

Comment (Aishwarya-U-R) [reply .github/workflows/ci-test-limited.yml:527]: So first 10 failing tests logs alone will be stored?

File: app/client/cypress/plugins/index.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ const chalk = require("chalk");
 // const _ = require("lodash");
 // const del = require("del");
 const cypressLogToOutput = require("cypress-log-to-output");
+const installLogsPrinter = require("cypress-terminal-report/src/installLogsPrinter");
 //const { isFileExist } = require("cy-verify-downloads");
```
Comment (KelvinOm) [inline app/client/cypress/plugins/index.js:10]: (nitpick) Could you clean up this useless comment?

**Review by KelvinOm — APPROVED** [review]

File: .github/workflows/ci-test-limited.yml (hunk 1 above)
Comment (akash-codemonk) [reply .github/workflows/ci-test-limited.yml:527]: Yup. 
This should be enough for now. We can change it as per requirement later on.

Comment (Aishwarya-U-R) [reply .github/workflows/ci-test-limited.yml:527]: Got it.

File: .github/workflows/ci-test-custom-script.yml [hunk 4]
```diff
@@ -472,6 +472,18 @@ jobs:
           working-directory: app/client
           env: "NODE_ENV=development"
 
+      - name: Trim number of cypress log files
+        if: failure()
```
Comment (Aishwarya-U-R) [inline .github/workflows/ci-test-custom-script.yml:476]: @sarojsarab - can you please review this? 
We need this step & below to be executed only when there are any failures in particular matrix

Comment (akash-codemonk) [reply .github/workflows/ci-test-custom-script.yml:476]: Using `faillure` seems to work fine. Run https://github.com/appsmithorg/appsmith/actions/runs/6405413689

Comment (Aishwarya-U-R) [reply .github/workflows/ci-test-custom-script.yml:476]: @sarojsarab - please confirm if using `failure()` is fine here.

Comment (sarojsarab) [reply .github/workflows/ci-test-custom-script.yml:476]: Yes it should be fine.

**Comment by Aishwarya-U-R** [comment]
@akash-codemonk - can this be merged now, so we can start capturing logs for analysis of this [long pending bug](https://github.com/appsmithorg/appsmith/issues/24174)?

**Comment by akash-codemonk** [comment]
> can this be merged now, so we can start capturing logs for analysis of this [long pending bug](https://github.com/appsmithorg/appsmith/issues/24174)?

Yes @Aishwarya-U-R we can merge it.

**Pull request merged**